src/hotspot/share/classfile/classFileParser.cpp

```diff
@@ -5963,7 +5963,11 @@ void ClassFileParser::mangle_hidden_class_name(InstanceKlass* const ik) {
   // Add a Utf8 entry containing the hidden name.
   assert(_class_name != NULL, "Unexpected null _class_name");
   int hidden_index = _orig_cp_size; // this is an extra slot we added
+
+PRAGMA_DIAG_PUSH
+PRAGMA_STRINGOP_OVERFLOW_IGNORED
   _cp->symbol_at_put(hidden_index, _class_name);
+PRAGMA_DIAG_POP
```

Comment:
`Array<T>::_data` is a pseudo flexible array member. "Pseudo" because C++
doesn't have flexible array members. The compiler is completely justified in
complaining about the apparently out-of-bounds accesses.

There is a "well-known" (though moderately ugly) approach to doing flexible
array members in C++. Something like this:

```
T* data() {
  return reinterpret_cast<T*>(
    reinterpret_cast<char*>(this) + data_offset());
}
```

where `data_offset()` is new and private:

```
static size_t data_offset() {
  return offset_of(Array, _data);
}
```

Use `data()` everywhere instead of using `_data` directly.

There are other places in HotSpot that use this kind of approach.

Reply:
Thanks @kimbarrett for your advice! Warnings from array.hpp have gone with your suggestion.

Reply:
Good.  I see you found and used the existing `base_offset_in_bytes` (which I'd overlooked), rather than using my suggested `data_offset`.